Plastic sheet production - Chemistry recipe

FantasticFwoosh opened the pull request:
:cl: Moonlighting Mac
Experiment: Due to budget cuts and oil synthesis replacing expensive processes of digging up haunted primeval ashwalker burial grounds, Nanotransen has taught chemists how to make plastic from chemicals in the hopes that new plastic products can reduce expenditure on metal and glass. 
add: you can now make plastic sheets from chemistry out of heated up crude oil, ash & sodium chloride
/:cl:

This Pr is atomised from what was originally planned but functionally it allows a methodical way of getting plastic within the game in a usually small quantity (100 units of formula = 10 sheets), making it over and over again would be tedious to discourage mass production and still keep some value to importing a lot of it if you need it.

Look forward to more plastic products in the future to compliment this.

FantasticFwoosh committed:
Adds plastic sheet crafting recipie

FantasticFwoosh committed:
Adds plastic polymer reagent

Nothing particularly fancy about it, toxic to consume (obviously not a great idea) compliments its crafting recipe which is far more interesting.

FantasticFwoosh committed:
Fixes syntax & errors

Now compiles and runs correctly, has been tested and works as expected (a little charcoal residue if there is leftover heated oil & ash but that's a nice touch) issues mostly down to my own silly little errors. 100 units in the reagent ratio = 10 sheets of plastic.

coiax commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -1546,6 +1546,18 @@
+	M.adjustToxLoss(1, 0)
+	. = 1
+	..()
+
```
So is this reagent ever actually made?

FantasticFwoosh replied:
It was a bit of a oversight post testing and realizing it was there. I left it in there in the case the recipe changes later (if people want to make glitter) or somehow people manage to cheat to get the reagent. It should be generic toxic anyway but its dead code not worth bothering about.

> Didn't quite realise the action to make the plastic sheet actually consumed it instantly. Oops.

coiax replied:
Don't put in reagents that are producible "just in case", just take it out. You literally described it as dead code.

FantasticFwoosh replied:
Well that's just my working excuse but since you mention it, sure ill just remove it before a maintainer blocks me with a crate of oranges requesting changes.

lordpidey commented:
I like it.  I especially like how it's possible to make some without using a chem dispenser. 

Burn paper for ash
Salt shaker from bar
Scoop up oil from maint

Iamgoofball commented on code/modules/reagents/chemistry/recipes/others.dm:
```diff
@@ -624,3 +624,14 @@
+
+/datum/chemical_reaction/plastic_polymers/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/i = 1, i <= created_volume, i++)
```
not how we do loops fam

FantasticFwoosh replied:
P.S - that's borrowed from the cheese making recipe, so your irks are probably demanded there. It works anyway.

> What exactly did i do wrong papa goof.

Iamgoofball replied:
# read the contributing guidelines

ExcessiveUseOfCobblestone replied:
don't be rude goof he is learning!

ExcessiveUseOfCobblestone replied:
if you are going to copypaste, use the Plasma solidification / Gold solidification as a base

FantasticFwoosh replied:
Its fine, be as rude as possible as long as its constructive. Its faster.

ExcessiveUseOfCobblestone replied:
wait this /IS/ exactly like what's in Plasma reaction, huh.

PAGING @RemieRichards for a lesson!

FantasticFwoosh replied:
Half of me is thinking Goof wants me to make between 10 pieces a reaction rather than 1 per small formula reagent dose as intended but that's just the paranoia talking. If that has any basis in truth Goof trust in me that the 1 per (5,2,3) ratio is deliberate.

Iamgoofball replied:
oh my god you idiot I want you to fix the for loop formatting to

# go by the guidelines I told you to read

Iamgoofball replied:
which you obviously didn't read by the way

# go read them

FantasticFwoosh replied:
I wish they had big font text like your posts.

Militaires replied:
>nanotren

dude my internet is so slow it cant load fucking github  and that's not a lie and painfully kept trying to log just to tell you how dumb of a mistake this is and this is not a joke

http://i.imgur.com/a6jSqQH.png

RemieRichards replied:
@FantasticFwoosh 
```for(var/i = 1, i <= created_volume, i++)```
->
```for(var/i in 1 to created_volume)```

FantasticFwoosh replied:
I noticed it too late @Militaires , ill fix it up.

@RemieRichards is this a explanation of the change or are you requesting that i should change it again because yours is more optimal?

FantasticFwoosh committed:
Consistency changes

Listening to feedback removing dead code for a chem that won't be seen, and a minor description change.

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -1546,6 +1546,13 @@
+/datum/reagent/plastic_polymers
+	name = "plastic polymers"
+	id = "plastic_polymers"
+	description = "the petroleum based components of plastic, contact a coder if you can actually see this" 
```
I want to say you can see this in the chem master so probably not best to have this breaking the meta wall.

coiax replied:
Although you shouldn't define an unobtainable reagent anyway

ExcessiveUseOfCobblestone replied:
oh lol yes if this isn't suppose to be made then don't have it :^)

FantasticFwoosh replied:
But you can't because the reagent is instantly consumed when it makes the plastic and is therefore inaccessible. If you can see it and have it stablised, its a bug. (unless im wrong)

ExcessiveUseOfCobblestone replied:
if I get my github functioning on my desktop i'll set this up for you bb

Iamgoofball replied:
you can see chem descriptions, yeah

FantasticFwoosh committed:
Fixes looping

As to conform with community guidelines. Personally as a coder freshman i have no idea how this works, but half of me is glad its more compact way to type out.

octareenroon91 commented:
Ash and sodium chloride make charcoal already

FantasticFwoosh commented:
Yes, but thats why you mix the oil first, divide or mix some oil to heat into ash, then add all the components to a cool beaker & heat up for plastic. It doesn't harshly overlap.

If you make uneven amounts, it can leave residue charcoal from a hot beaker: Intended.

octareenroon91 commented:
Then you're fighting a race condition. It's sensitive to whatever the DM and DD feel like doing.

lordpidey commented:
Hmm, it might be best to bump up the required temp for charcoal then.  Right now it's at 380, which is very close to 376.

FantasticFwoosh commented:
No its not, stop fretting.

If you put in the exact reagent formula it makes, the oil forces the plastic recipe first. (5 oil, 2 salt, 3 ash) rather than the ash & salt together in heat & add oil (because that illogical to heat up salt), in testing practice it's been consistent with my mixing method all the while.

KorPhaeron commented:
What can plastic even be used for?

Converting our infinite energy into matter never ends well

FantasticFwoosh commented:
Right now plastic flaps & plastic golems (merged prior this post) but i plan to expand upon the system more and make it more widespread, covering some holes in production of objects.

> spray bottles, blood bags etc.

FantasticFwoosh committed:
Rectifies description
